worlds/kh2/__init__.py

```diff
@@ -23,99 +33,124 @@ class KingdomHearts2Web(WebWorld):
+        # slot_data_duping:Set[str)
+        # random_super_boss_list:List[str)
+        # growth_list:List[str)
+        # total_location:int
+        self.sora_ability_dict = None
+        self.goofy_ability_dict = None
+        self.donald_ability_dict = None
+        self.slot_data_duping = set()
+        self.random_super_boss_list = list()
         self.growth_list = list()
+        self.total_locations = len(all_locations.keys())
         for x in range(4):
             self.growth_list.extend(Movement_Table.keys())
-        self.slotDataDuping = set()
-        self.localItems = dict()
 
     def fill_slot_data(self) -> dict:
-        for values in CheckDupingItems.values():
-            if isinstance(values, set):
-                self.slotDataDuping = self.slotDataDuping.union(values)
-            else:
-                for inner_values in values.values():
-                    self.slotDataDuping = self.slotDataDuping.union(inner_values)
-        self.LocalItems = {location.address: item_dictionary_table[location.item.name].code
-                           for location in self.multiworld.get_filled_locations(self.player)
-                           if location.item.player == self.player
-                           and location.item.name in self.slotDataDuping
-                           and location.name not in AllWeaponSlot}
-
-        return {"hitlist":              self.hitlist,
-                "LocalItems":           self.LocalItems,
-                "Goal":                 self.multiworld.Goal[self.player].value,
-                "FinalXemnas":          self.multiworld.FinalXemnas[self.player].value,
-                "LuckyEmblemsRequired": self.multiworld.LuckyEmblemsRequired[self.player].value,
-                "BountyRequired":       self.multiworld.BountyRequired[self.player].value}
-
-    def create_item(self, name: str, ) -> Item:
-        data = item_dictionary_table[name]
-        if name in Progression_Dicts["Progression"]:
+        for ability in self.slot_data_sora_weapon:
+            if ability in self.sora_ability_dict.keys():
+                if self.sora_ability_dict[ability] >= 1:
+                    self.sora_ability_dict[ability] -= 1
+        self.donald_ability_dict = {k: v.quantity for k, v in DonaldAbility_Table.items()}
+        for ability in self.slot_data_donald_weapon:
+            if ability in self.donald_ability_dict.keys():
+                if self.donald_ability_dict[ability] >= 1:
+                    self.donald_ability_dict[ability] -= 1
+        self.goofy_ability_dict = {k: v.quantity for k, v in GoofyAbility_Table.items()}
+        for ability in self.slot_data_goofy_weapon:
+            if ability in self.goofy_ability_dict.keys():
+                if self.goofy_ability_dict[ability] >= 1:
+                    self.goofy_ability_dict[ability] -= 1
+        return {
+            "hitlist":                [], # remove this after next update
```

Comment:
because it does .lower in the function it doesnt work with the client, I could change the client to take both or I could just keep it the way it is since its simpler

Reply:
as_dict has a `casing` kwarg that changes the output keys

Reply:
camel has split_name[0].lower()

Reply:
oh wait im using pascal

Reply:
fwiw the default returns the attribute with the exact same casing as defined in the class body

Reply:
so when using pascal I input
self.options.as_dict("Goal","FinalXemnas","LuckyEmblemsRequired","BountyRequired",casing="pascal")
and its returning
{'Goal': 1, 'Finalxemnas': 1, 'Luckyemblemsrequired': 29, 'Bountyrequired': 18}
which is not pascal since the original are pascal

Reply:
i suppose i could make it attempt to split from capital letters, since it currently assumes the fields are snake_cased, but if your original is already pascal you shouldn't need the kwarg at all.